app/src/main/java/com/kickstarter/viewmodels/ProjectCardHolderViewModel.kt

```diff
@@ -0,0 +1,559 @@
+        override fun projectTagContainerIsGone(): Observable<Boolean> = projectTagContainerIsGone
+        override fun projectWeLoveIsGone(): Observable<Boolean> = projectWeLoveIsGone
+        override fun rootCategoryNameForFeatured(): Observable<String> = rootCategoryNameForFeatured
+        override fun setDefaultTopPadding(): Observable<Boolean> = setDefaultTopPadding
+        override fun savedViewGroupIsGone(): Observable<Boolean> = savedViewGroupIsGone
+        override fun notifyDelegateOfHeartButtonClicked(): Observable<Project> = this.notifyDelegateOfHeartButtonClicked
+
+        init {
+            projectForDeadlineCountdownDetail = project
+            backersCountTextViewText = project
+                .map { it?.backersCount() }
+                .map { value ->
+                    value?.let { it ->
+                        NumberUtils.format(
+                            it
+                        )
+                    }
+                }
+
+            backingViewGroupIsGone = project
+                .map { it?.metadataForProject() !== ProjectMetadata.BACKING }
+
+            deadlineCountdownText = project
+                .map { it.deadlineCountdownValue() }
+                .map { value: Int? ->
+                    NumberUtils.format(
+                        value!!
+                    )
+                }
+
+            project
+                .map { p -> if (p.isStarred()) R.drawable.icon__heart else R.drawable.icon__heart_outline }
+                .subscribe(this.heartDrawableId)
+
+            project
+                .compose(Transformers.takeWhen(heartButtonClicked))
+                .filter { ObjectUtils.isNotNull(it) }
+                .map { requireNotNull(it) }
+                .compose(bindToLifecycle())
+                .subscribe { notifyDelegateOfHeartButtonClicked.onNext(it) }
+
+            featuredViewGroupIsGone = project
+                .map { it?.metadataForProject() !== ProjectMetadata.CATEGORY_FEATURED }
+
+            friendAvatarUrl1 = project
+                .filter(Project::isFriendBacking)
+                .filter { ObjectUtils.isNotNull(it) }
+                .map { requireNotNull(it) }
+                .map { it.friends() }
+                .map { it[0].avatar().small() }
+
+            friendAvatarUrl2 = project
+                .filter(Project::isFriendBacking)
+                .filter { ObjectUtils.isNotNull(it) }
+                .map { requireNotNull(it) }
+                .map { it.friends() }
+                .filter { it.size > 1 }
+                .map { it[1].avatar().small() }
+
+            friendAvatarUrl3 = project
+                .filter(Project::isFriendBacking)
+                .filter { ObjectUtils.isNotNull(it) }
+                .map { requireNotNull(it) }
+                .map { it.friends() }
+                .filter { it.size > 2 }
+                .map { it[2].avatar().small() }
```

Comment:
For the avatar images, filter as well in case the url is empty. Getting this error when testing. 
<img width="1201" alt="Screen Shot 2022-02-07 at 3 20 53 PM" src="https://user-images.githubusercontent.com/4083656/152889762-89ba8c7e-f182-4e22-add6-b575a5d0bbdb.png">

Probably nice to have as well on the `ProjectCardViewHolder` com.kickstarter.ui.viewholders.ProjectCardViewHolder#setFriendAvatarUrl we can use some of the image extensions we have instead of calling apollo directly

Reply:
@Arkariang 
![image](https://user-images.githubusercontent.com/1075310/152994189-e6dc432c-8a98-4b23-9899-2596502330cd.png)
Created this method
`fun Avatar.replaceSmallImageWithMediumIfEmpty(): String = this.small().ifEmpty { this.medium() }`